zookeeper-api/src/main/java/org/apache/helix/zookeeper/datamodel/serializer/ZNRecordSerializer.java

```diff
@@ -86,7 +98,8 @@ private static int getListFieldBound(ZNRecord record) {
       mapper.writeValue(baos, data);
       serializedBytes = baos.toByteArray();
       // apply compression if needed
-      if (record.getBooleanField("enableCompression", false) || serializedBytes.length > ZNRecord.SIZE_LIMIT) {
+      if (record.getBooleanField("enableCompression", false) || (autoCompressionEnabled
+          && serializedBytes.length > autoCompressionThreshold)) {
```

Comment:
ZnRecordStreamingSerializer also has this logic. It would be great if you could refactor this piece of logic out so that we support this in all of our ZNRecord serializers :)

Reply:
Right already realized that. Just need to confirm with the team on this, as I was initially working on ZNRecordSerializer. But also found the same logic in ZnRecordStreamingSerializer.

Reply:
This logic could go to the ZNRecord class. isAutoCompression()